### PR TITLE
Add new module : Tcoffee irmsd

### LINKED ISSUE #4022: new module: tcoffee_irmsd
### Is there an existing module for this?

- [X] I have searched for the existing module

### Is there an open PR for this?

- [X] I have searched for existing PRs

### Is there an open issue for this?

- [X] I have searched for existing issues

### Are you going to work on this?

- [X] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by luisas**
## PR checklist

Add new module tcoffee irmsd 

Closes #4022 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/tcoffee/irmsd/main.nf [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        'https://depot.galaxyproject.org/singularity/t-coffee:13.46.0.919e8c6b--hfc96bf3_0':
+        'biocontainers/t-coffee:13.46.0.919e8c6b--hfc96bf3_0' }"
+
+    input:
+    tuple  val(meta),  file (msa)
+    tuple  val(meta2), file(template), file(structures)
+
+    output:
+    tuple val(meta), path ("${prefix}.irmsd"), emit: irmsd
+    path "versions.yml" , emit: versions
```
Comment (JoseEspinosa) [inline modules/nf-core/tcoffee/irmsd/main.nf:17]:
```suggestion
    tuple val(meta), path ("${prefix}.irmsd"), emit: irmsd
    path "versions.yml"                      , emit: versions
```

File: modules/nf-core/tcoffee/irmsd/main.nf [hunk 2]
```diff
@@ -0,0 +1,50 @@
+
```
Comment (JoseEspinosa) [inline modules/nf-core/tcoffee/irmsd/main.nf:1]:
```suggestion
```

File: modules/nf-core/tcoffee/irmsd/main.nf [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    label 'process_medium'
+
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/t-coffee:13.46.0.919e8c6b--hfc96bf3_0':
+        'biocontainers/t-coffee:13.46.0.919e8c6b--hfc96bf3_0' }"
+
+    input:
+    tuple  val(meta),  file (msa)
+    tuple  val(meta2), file(template), file(structures)
```
Comment (JoseEspinosa) [inline modules/nf-core/tcoffee/irmsd/main.nf:13]: I don't see structures used anywhere, are they just read by tcoffee if they are found in the path?

File: modules/nf-core/tcoffee/irmsd/meta.yml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+      type: file
+      description: fasta file containing the alignment to be evaluated
+      pattern: "*.{aln,fa,fasta,fas}"
+  - template:
+      type: file
+      description: template file matching the structures to the sequences in the alignment
+      pattern: "*"
+  - structures:
+      type: directory
+      description: structures file matching the sequences in the alignment
```
Comment (JoseEspinosa) [inline modules/nf-core/tcoffee/irmsd/meta.yml:32]:
```suggestion
      description: Directory containing the structures file matching the sequences in the alignment in PDB format
```

File: modules/nf-core/tcoffee/irmsd/meta.yml [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+      description: |
+        Groovy Map containing sample information
+        e.g. [ id:'test', ... ]
+  - msa:
+      type: file
+      description: fasta file containing the alignment to be evaluated
+      pattern: "*.{aln,fa,fasta,fas}"
+  - template:
+      type: file
+      description: template file matching the structures to the sequences in the alignment
```
Comment (JoseEspinosa) [inline modules/nf-core/tcoffee/irmsd/meta.yml:28]:
```suggestion
      description: Template file matching the structures to the sequences in the alignment
```

File: modules/nf-core/tcoffee/irmsd/meta.yml [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+      description: |
+        Groovy Map containing sample information
+        e.g. [ id:'test', single_end:false ]
+  - versions:
+      type: file
+      description: File containing software versions
+      pattern: "versions.yml"
+  - irmsd:
+      type: file
+      description: a file containing the irmsd of the alignment
```
Comment (JoseEspinosa) [inline modules/nf-core/tcoffee/irmsd/meta.yml:47]:
```suggestion
      description:  File containing the irmsd of the alignment
```

File: modules/nf-core/tcoffee/irmsd/tests/main.nf.test [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+                input[1] = UNTAR.out.untar.map { meta,dir -> [[ id:'test' ], file("https://raw.githubusercontent.com/nf-core/test-datasets/multiplesequencealign/testdata/templates/seatoxin-ref_template.txt", checkIfExists: true) ,file(dir).listFiles().collect()]}
+                """
+            }
+            
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert path(process.out.irmsd.get(0).get(1)).getText().contains("1ahl") }            )
```
Comment (JoseEspinosa) [inline modules/nf-core/tcoffee/irmsd/tests/main.nf.test:44]:
```suggestion
                { assert path(process.out.irmsd.get(0).get(1)).getText().contains("1ahl"))
```

File: tests/config/pytest_modules.yml [hunk 8]
```diff
@@ -1,2810 +1,3737 @@
 abritamr/run:
```
Comment (JoseEspinosa) [inline tests/config/pytest_modules.yml:1]: Why this file is changed?

**Review by JoseEspinosa — CHANGES_REQUESTED** [review]
Nice!
I suggested some changes and asked a couple of questions

File: modules/nf-core/tcoffee/irmsd/main.nf (hunk 3 above)
Comment (luisas) [reply modules/nf-core/tcoffee/irmsd/main.nf:12]: Yes! Via the template, tcoffee knows which ones to use, and looks in the same directory. 

As far as I know, this is the standard way coffee works and did not find any way to explicitly point to the structure files.

File: tests/config/pytest_modules.yml (hunk 8 above)
Comment (luisas) [reply tests/config/pytest_modules.yml:1]: Good question! Idk!!! 

I did not notice it, i will revert it!

**Review by suzannejin — APPROVED** [review]

**Review by JoseEspinosa — APPROVED** [review]
Awesome! :rocket:

**Pull request merged**